### PR TITLE
Updated to yaml cpp version with less compiler warnings and fixed patch.

### PR DESCRIPTION
**Pull request by andrepfr**
Ptal

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -6,11 +6,13 @@ catkin_simple()
 
 include(ExternalProject)
 
+set(YAML_CPP_TAG 11607eb5bf1258641d80f7051e7cf09e317b4746)
```
Comment (schneith) [inline CMakeLists.txt]: would it be possible to use a tag here? e.g. yaml-cpp-0.5.3
that way it would be easier to figure out the version

Comment (andrepfr) [inline CMakeLists.txt]: Unfortunately all releases seem to be not using C++11 features. So they still contain std::auto_ptr which creates a warning. According to this [link](https://github.com/jbeder/yaml-cpp/releases/tag/yaml-cpp-0.5.3) the next release should have this fixed. But for now I would suggest to leave it. Wdyt?

Comment (schneith) [inline CMakeLists.txt]: Ok, sounds good. But could you please add a comment for L9 describing the reasons why we fix to a non-release commit?

Comment (andrepfr) [inline CMakeLists.txt]: Sure, will add a comment. I just wanted to get rid of those std::auto_ptr compiler warnings.

Comment (schneith) [inline CMakeLists.txt]: Cool, now we are one small step closer to use -Werror :)
Thanks for the fixes.

**Pull request merged**